mav_local_planner/src/mav_local_planner.cpp

```diff
@@ -591,8 +591,9 @@ void MavLocalPlanner::commandPublishTimerCallback(
   constexpr size_t kQueueBuffer = 0;
 
   // publish current progress in percentage
-  progress_pub_.publish(
-      (int)(float(path_index_) / float(path_queue_.size()) * 100));
+  std_msgs::Int8 progress_msg;
+  progress_msg.data = (int8_t) (float(path_index_) / float(path_queue_.size()) *100);
```

Comment:
These C style casts should basically never be used in C++ is my understanding. Replace with static_cast.

There's a good argument for this somewhere...